src/Serialize.h

```diff
@@ -239,7 +239,7 @@ namespace drafter {
         }
 
         template<typename Collection, typename Functor, typename Arg1, typename Arg2>
-        R operator()(const Collection& collection, Functor &wrapper, Arg1 arg1, Arg2 arg2) const {
+        R operator()(const Collection& collection, Functor &wrapper, Arg1 arg1, Arg2& arg2) const {
```

Comment:
Won't work. We need the first arg1 to be by value since it was passed pointers for the payload sections. And yes, its necessary and no, we can't change that.